_posts/2021-05-21-js-waku.md

```diff
@@ -0,0 +1,171 @@
+2021-05-21-js-waku.md---
+layout: post
+name:  "Presenting JS-Waku: Waku v2 in the browser"
+title:  "Presenting JS-Waku: Waku v2 in the browser"
+date:   2021-05-21 12:00:00 +0800
+author: franck
+published: true
+permalink: /2021-05-21-js-waku
+categories: platform
+summary: "We are summarizing the Waku v2 status, introducing JS-Waku, what we achieved so far and what is next."
+discuss: 
+---
+
+After almost three months working on JS-Waku,
+we thought it would be a good time to give it a proper introduction.
+
+## Waku v2
+
+First, let's review what is Waku v2.
+
+Waku v2 comes from a need to have a more scalable, better optimised solution for the Status app to achieve decentralised
+communications on resource restricted devices (ie, phones).
+
+The Status chat feature was initially built over Whisper.
+However, Whisper has a number of caveats which makes it inefficient for mobile phones.
+For example, with Whisper, all devices are receiving all messages which is not ideal for limited data plans.
+
+To remediate this, a Waku mode (then Waku v1), based on devp2p, was introduced.
+To further enable web and restricted resource devices, Waku v2 has been created based on libp2p.
+The migration of the Status chat feature to Waku v2 is currently in progress.
+
+We see the need of such solution in the broader Ethereum ecosystem, beyond Status.
+This is why are building Waku v2 as a decentralised communication platform for all to use and build on.
+If you want to read more about Waku v2 and what it aims to achieve,
+checkout [What's the Plan for Waku v2?](/waku-v2-plan).
+
+Since last year, we have been busy defining and implementing Waku v2 protocols in [nim-waku](https://github.com/status-im/nim-waku),
+from which you can build [wakunode2](https://github.com/status-im/nim-waku#wakunode).
+Wakunode2 is an adaptive and modular Waku v2 node,
+it allows users to run their own node and use the Waku v2 protocols they need.
+The nim-waku project doubles as a library, that can be used to add Waku v2 support to native applications.
+
+## Waku v2 in the browser
+
+We believe that dApps and wallets can benefit from the Waku network in several ways.
+For some dApps, it makes sense to enable peer-to-peer communications.
+For others, machine-to-machine communications would be a great asset.
+For example, in the case of a DAO,
+Waku could be used for gas-less voting.
+Enabling the DAO to notify their users of a new vote,
+and users to vote without interacting with the blockchain and spending gas.
+
+[Murmur](https://github.com/status-im/murmur) was the first attempt to bring Waku to the browser,
+acting as a bridge between Whisper and Waku.
+Then, a web chat [POC](https://github.com/vacp2p/waku-web-chat) was created to demonstrate the potential of Waku v2
```

Comment:
Once Waku v2 was started and there was a native implementation on top of libp2p, this proof of concept was craeted